.github/workflows/build-test-osx-m1.yaml

```diff
@@ -7,41 +7,60 @@ name: build-test-osx-m1
 
 on:
   workflow_dispatch:
+    inputs:
+      use-cache:
+        description: "Use Opam Cache - uncheck the box to disable use of the opam cache, meaning a long-running but completely from-scratch build."
+        required: true
+        type: boolean
+        default: true
   workflow_call:
+    inputs:
+      use-cache:
+        description: "Use Opam Cache - uncheck the box to disable use of the opam cache, meaning a long-running but completely from-scratch build."
+        required: false
+        type: boolean
+        default: true
 
 jobs:
   build-core-m1:
     name: Build the OSX M1 binaries
-    #TODO? if the use of the GHA-hosted runner works well in build-test-osx-x86.yaml,
-    # we might as well move out of a self-hosted runner for M1 too.
-    # This would remove the need for those ugly hard cleanup below because
-    # our self-hosted runner builds are not "hermetic".
-    runs-on: [self-hosted, macOS, ARM64, aws]
+    runs-on:
+      [
+        "self-hosted",
+        "macOS",
+        "ARM64",
+        "ghcr.io/cirruslabs/macos-monterey-xcode:latest",
+      ]
     env:
       OPAM_SWITCH_NAME: "4.14.0"
     steps:
-      # Force hard cleanup of any old checkouts. This is needed because we run on
-      # self-hosted CI runners here and when submodules move around they are no longer
-      # removed by default.
-      # This hack misses some files (like hidden files) but should take care of the
-      # issues we've been seeing.
-      - name: Clean Checkout
+      - name: Setup runner directory
```

Comment:
Can you add a comment explaining why we now need this extra setup? Is it because we use this Cirrus thing?
Is there a link to a doc explaining the need for this setup?